### PR TITLE
Scope ResourceTypeFilters and ResourceTypeSelector

### PR DESCRIPTION
**Pull request by gilhsu**
~~On top of #120~~
Scope ResourceTypeFilters and ResourceTypeSelector to what is saved or marked in collection, depending on modal filters.

https://user-images.githubusercontent.com/45667486/114898409-0fe78b80-9de0-11eb-9761-3a4f995b7d7d.mov

### CODE REVIEW

File: src/redux/selectors/index.js [hunk 1]
```diff
@@ -112,24 +112,28 @@ export const allValidRecordsSortedByDateSelector = createSelector(
     .sort(sortByDate),
 );
 
-export const filteredRecordsSelector = createSelector(
+export const recordsWithShowOnlyFiltersSelector = createSelector(
```
Comment (gilhsu) [inline src/redux/selectors/index.js]: need to separate `showOnly` filters so this selector can be used with `resourceTypeFiltersSelector`

File: src/redux/selectors/index.js [hunk 2]
```diff
@@ -208,13 +212,25 @@ export const patientAgeAtResourcesSelector = createSelector(
...
+      .reduce((acc, { type }) => {
+        if (acc[type] === undefined) {
+          acc[type] = resourceTypeFilters[type];
+        }
+        return acc;
+      }, {});
+
+    // sorting
+    return Object.keys(filteredResourceTypeFilters)
+      .sort()
```
Comment (foxtree) [inline src/redux/selectors/index.js:247]: is sorting by these `key`s legitimate?

...there's an Object being returned

File: src/components/FilterDrawer/FilterDrawer.js [hunk 3]
```diff
@@ -53,13 +53,13 @@ ResourceTypeFilter.propTypes = {
...
   toggleResourceTypeFilterAction,
-  orderedResourceTypeFilters,
+  resourceTypeFilters,
   children,
 }) => {
   const renderDrawer = () => (
     <View>
       <Text style={styles.drawerTitle}>Resource Type Filters</Text>
-      {Object.entries(orderedResourceTypeFilters).map(([resourceType, value]) => (
+      {Object.entries(resourceTypeFilters).map(([resourceType, value]) => (
```
Comment (foxtree) [inline src/components/FilterDrawer/FilterDrawer.js:62]: sorting by label should occur here

Comment (foxtree) [reply src/components/FilterDrawer/FilterDrawer.js:62]: sorting could occur in the selector, and it returns an Array of Objects, maybe:
`[{ type, label, enabled }, { type, label, enabled }, { type, label, enabled }]`

**Pull request closed without merging**